tracking_allocator.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
```

Comment:
pragma once won't work properly on some filesystems (distributed cloud stuff, errors are very hard to diagnose). Can we stick to include guards?

Reply:
We (I) should set up clang-tidy to pick this up.